src/adapter/src/coord.rs

```diff
@@ -1442,6 +1622,48 @@ impl Coordinator {
     ) -> Result<(), AdapterError> {
         info!("coordinator init: beginning bootstrap");
 
+        // Initialize cluster replica statuses.
+        // Gross iterator is to avoid partial borrow issues.
+        let cluster_statuses: Vec<(_, Vec<_>)> = self
+            .catalog()
+            .clusters()
+            .map(|cluster| {
+                (
+                    cluster.id(),
+                    cluster
+                        .replicas()
+                        .map(|replica| {
+                            (replica.replica_id, replica.config.location.num_processes())
+                        })
+                        .collect(),
+                )
+            })
+            .collect();
+        for (cluster_id, replica_statuses) in cluster_statuses {
+            self.cluster_replica_statuses
+                .initialize_cluster_statuses(cluster_id);
+            for (replica_id, num_processes) in replica_statuses {
+                self.cluster_replica_statuses
+                    .initialize_cluster_replica_statuses(
+                        cluster_id,
+                        replica_id,
+                        num_processes,
+                        self.now_datetime(),
```

Comment:
This seems lightly sketchy. Does this timestamp matter?

Reply:
What specifically is sketchy about it? I just pulled the timestamp out of the loop so that they're all the same if that's what you meant.

Reply:
That we are using the wall time at all. It's fine, I see that we need it.

Reply:
Oh yeah, this whole table is a bit sketchy in terms of linearizability. As mentioned it should probably be a `BuiltinSource` and not a `BuiltinTable` at all. FWIW, this is the same as the existing behavior, we reset the status of all replicas at startup to the not ready with the current wall time.